rtos/Thread.h

```diff
@@ -317,13 +317,28 @@ class Thread : private mbed::NonCopyable<Thread> {
     */
     static osEvent signal_wait(int32_t signals, uint32_t millisec=osWaitForever);
 
-    /** Wait for a specified time period in millisec:
+    /** Wait for a specified time period in milliseconds
+      Being tick-based, the delay will be up to the specified time - eg for
+      a value of 1 the system waits until the next millisecond tick occurs,
+      leading to a delay of 0-1 milliseconds.
       @param   millisec  time delay value
       @return  status code that indicates the execution status of the function.
       @note not callable from interrupt
+      @note this call is equivalent to Kernel::delay()
     */
     static osStatus wait(uint32_t millisec);
 
+    /** Wait until a specified time in millisec
+      The specified time is according to Kernel::get_ms_count().
+      @param   millisec absolute time in millisec
+      @return  status code that indicates the execution status of the function.
+      @note not callable from interrupt
+      @note if the time in the past, or >= 2^32-1 ms in the future, this
```

Comment:
This will need to be updated if the tick API changes to 32 bit in RTX (as mentioned above).